### ISSUE TITLE
Can not reproduce the pretrained model

### ISSUE THREAD

**Issue by EDENpraseHAZARD** [ClementPinard/SfmLearner-Pytorch#101]
Hi Clement, recently I trained the model used the arguments you gave, but got bad result.

![loss](https://user-images.githubusercontent.com/53077461/87124258-06624d80-c2bb-11ea-9fe3-dcd377b5cd4e.PNG)

I checked the tensorboard and found that the disparity_smoothness_loss is different from yours.

![tensorboard](https://user-images.githubusercontent.com/53077461/87124535-8dafc100-c2bb-11ea-80f3-a265c5cc51b6.PNG)

Then I noticed the factor of smooth loss and also tried the loss divided by 2.0, but the result was still bad .The disparity_smoothness_loss was still increasing.

```
python3 ./train.py \
        /gdata/gaoyuan/SfMLearner_kitti/kitti/ \
        -b 4 \
        -m 0 \
        -s 0.2 \
        --epoch-size 1000  \
        --sequence-length 5 \
        --log-output \
        --with-gt
```
The arguments I used is absolutely the same as yours and the image I used is pytorch/pytorch:1.5-cuda10.1-cudnn7-devel which may not affect training. What's wrong with my training?

**Comment by ClementPinard** [comment]
Hi,

the model was trained a wile ago (in 2018), and maybe some regression has been introduced, either from my part or from pytorch which interface and default options may have changed. I'll try to investigate it in the near future when I have time

In the mean time, the tweaks I had to do in order to make it work as good as the original article in tensroflow (because just a code translation with the same hyperparameters would not work) all resided in two main factors :
 - early stopping : results might worsen after 100k iterations, hence the use of validation set, in order to stop the training at the right time. Notice that depth accuracy decreases before pose accuracy, which means in your case, you might want to train for longer.
 - Sooth loss tweaking : this one was tough ! The original 0.5 smooth weight made my network go completely flat, it was apparently to important compared to photometric loss. I tried to divide by a bit more than 2 at each scale (e.g. 2.8) (see [this part of the code](https://github.com/ClementPinard/SfmLearner-Pytorch/blob/master/loss_functions.py#L87) , and tried to apply it on depth instead of disp. By experience, I happened to know that the right training usually begins with a smooth loss that goes to almost 0, and then starts to rise again. Additionally, smaller depth scales usually get better sooner than larger scales.
